backend/main-service/src/main/java/kkakka/mainservice/product/ui/ProductController.java

```diff
@@ -1,9 +1,69 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
 
 @RestController
-@RequestMapping("/product")
+@RequestMapping("/products")
 public class ProductController {
+
+    private final ProductRepository productRepository;
+    private final ProductService productService;
+    private final CategoryService categoryService;
+
+    public ProductController(ProductRepository productRepository, ProductService productService
+            , CategoryService categoryService) {
+        this.productRepository = productRepository;
+        this.productService = productService;
+        this.categoryService = categoryService;
+    }
+
+    @GetMapping("/health_check")
+    public String status() {
+        return "It's Working in Product Service";
+    }
+
+    @PostConstruct
+    public void init() {
+
+        Category category = new Category(1L, "비스킷/샌드");
+        productRepository.save(new Product(category, "롯데 제로 초콜릿칩 쿠키 168g", 4480, 10, "https://user-images.githubusercontent.com/99088509/191633507-6280963f-6363-4137-ac2a-a8a060d28669.png", "상세URL"));
+        productRepository.save(new Product(category, "롯데 제로 초콜릿칩 쿠키 168g", 4480, 10, "https://user-images.githubusercontent.com/99088509/191633507-6280963f-6363-4137-ac2a-a8a060d28669.png", "상세URL"));
+        productRepository.save(new Product(category, "롯데 제로 초콜릿칩 쿠키 168g", 4480, 10, "https://user-images.githubusercontent.com/99088509/191633507-6280963f-6363-4137-ac2a-a8a060d28669.png", "상세URL"));
+        productRepository.save(new Product(category, "롯데 제로 초콜릿칩 쿠키 168g", 4480, 10, "https://user-images.githubusercontent.com/99088509/191633507-6280963f-6363-4137-ac2a-a8a060d28669.png", "상세URL"));
+        productRepository.save(new Product(category, "롯데 제로 초콜릿칩 쿠키 168g", 4480, 10, "https://user-images.githubusercontent.com/99088509/191633507-6280963f-6363-4137-ac2a-a8a060d28669.png", "상세URL"));
+    }
+
+    @GetMapping("/{productId}")
+    public ResponseEntity<ProductResponseDto> showProductDetail(@PathVariable("productId") Long productId) {
+
+        ProductResponseDto responseDto = productService.getProductDetail(productId);
+        return ResponseEntity.status(HttpStatus.OK).body(responseDto);
+    }
+
+    @GetMapping("/products")
+    public ResponseEntity<List<ResponseCategoryProducts>> showCategoryProducts(@RequestParam("category") Long categoryId,
+                                                                        Pageable pageable) {
+        Page<ResponseCategoryProducts> result = categoryService.getProductsByCategoryId(categoryId, pageable);
+
+        System.out.println("전체 페이지 수: " + result.getTotalPages());
+        System.out.println("전체 상품목록 수: " + result.getTotalElements());
+        System.out.println("현재 페이지(기본0) : " + result.getNumber());
+        System.out.println("hasNext: " + result.hasNext());
+
```

Comment:
```suggestion
    // Logger 선언
    // Logger log = LoggerFactory.getLogger(ProductController.class);

        log.info("전체 페이지 수: {}", result.getTotalPages());
        log.info("전체 상품목록 수: {}", result.getTotalElements());
        log.info("현제 페이지(기본0): ", result.getNumber());
        log.info("hasNext: {}", result.hasNext());
```

로그로 꼭 남겨야 하는 정보는 저라면 이렇게 남길 것 같네요ㅎㅎ

Reply:
산넘어산으로 해결할게 산더미네요 ㅜ 피드백 감사합니다